codegen/writer/writer.go

```diff
@@ -49,7 +49,9 @@ func (w *DefaultFileWriter) WriteFiles(files []render.OutFile) error {
 			commentPrefix = "#"
 		}
 
-		if w.Header != "" {
+		if file.HeaderOverride != nil {
```

Comment:
nit: couldn't we disable the header based on file type? I am fine with it as is too.

Reply:
yeah for sure good call, I agree w/ that in hindsight

At the time I thought this HeaderOverride solution might be more flexible but given that this one json file is the only use case and the added complexity I'm inclined to just disable header for json as well. it aligns better w/ the current pattern of choosing header by file type.

Reply:
fixed, I removed this